File: bet_dashboard/frontend/src/config/marketConfig.ts
/**
 * Single source of truth for market column configuration.
 * Reorder entries here to change column order in the betting tips table.
 */

export interface MarketColumn {
    market: string;        // Market identifier (e.g., '1', 'X', 'Over 2.5')
    label: string;         // Display label in table header
    consKey: string;       // Key for consensus value in Match object
    oddsKey: string;       // Key for odds value in Match object
    marketType: string;    // Market type for backend (e.g., 'result', 'over_under_25')
}

/**
 * Market columns in display order.
 * To reorder columns, simply reorder entries in this array.
 */
export const MARKET_COLUMNS: MarketColumn[] = [
    { market: '1', label: '1', consKey: 'cons_home', oddsKey: 'odds_home', marketType: 'result' },
    { market: 'X', label: 'X', consKey: 'cons_draw', oddsKey: 'odds_draw', marketType: 'result' },
    { market: '2', label: '2', consKey: 'cons_away', oddsKey: 'odds_away', marketType: 'result' },
    { market: 'Over 2.5', label: 'O2.5', consKey: 'cons_over_25', oddsKey: 'odds_over_25', marketType: 'over_under_25' },
    { market: 'Under 2.5', label: 'U2.5', consKey: 'cons_under_25', oddsKey: 'odds_under_25', marketType: 'over_under_25' },
    { market: 'BTTS Yes', label: 'BTTS Y', consKey: 'cons_btts_yes', oddsKey: 'odds_btts_yes', marketType: 'btts' },
    { market: 'BTTS No', label: 'BTTS N', consKey: 'cons_btts_no', oddsKey: 'odds_btts_no', marketType: 'btts' },
    { market: 'Over 0.5', label: 'O0.5', consKey: 'cons_over_05', oddsKey: 'odds_over_05', marketType: 'over_under_05' },
    { market: 'Under 0.5', label: 'U0.5', consKey: 'cons_under_05', oddsKey: 'odds_under_05', marketType: 'over_under_05' },
    { market: 'Over 1.5', label: 'O1.5', consKey: 'cons_over_15', oddsKey: 'odds_over_15', marketType: 'over_under_15' },
    { market: 'Under 1.5', label: 'U1.5', consKey: 'cons_under_15', oddsKey: 'odds_under_15', marketType: 'over_under_15' },
    { market: 'Over 3.5', label: 'O3.5', consKey: 'cons_over_35', oddsKey: 'odds_over_35', marketType: 'over_under_35' },
    { market: 'Under 3.5', label: 'U3.5', consKey: 'cons_under_35', oddsKey: 'odds_under_35', marketType: 'over_under_35' },
    { market: 'Over 4.5', label: 'O4.5', consKey: 'cons_over_45', oddsKey: 'odds_over_45', marketType: 'over_under_45' },
    { market: 'Under 4.5', label: 'U4.5', consKey: 'cons_under_45', oddsKey: 'odds_under_45', marketType: 'over_under_45' },
    { market: '1X', label: '1X', consKey: 'cons_dc_1x', oddsKey: 'odds_dc_1x', marketType: 'double_chance' },
    { market: '12', label: '12', consKey: 'cons_dc_12', oddsKey: 'odds_dc_12', marketType: 'double_chance' },
    { market: 'X2', label: 'X2', consKey: 'cons_dc_x2', oddsKey: 'odds_dc_x2', marketType: 'double_chance' },
];

/**
 * Table column definition for headers.
 */
export interface TableColumn {
    key: string;
    label: string;
    wide?: boolean;
}

/**
 * Fixed columns that appear before market columns.
 */
export const FIXED_COLUMNS: TableColumn[] = [
    { key: 'datetime', label: 'Date' },
    { key: 'home', label: 'Home', wide: true },
    { key: 'away', label: 'Away', wide: true },
    { key: 'sources', label: 'Sources' },
];

/**
 * Generate full column list for table headers.
 */
export function getTableColumns(): TableColumn[] {
    return [
        ...FIXED_COLUMNS,
        ...MARKET_COLUMNS.map(m => ({ key: m.consKey, label: m.label })),
    ];
}

/**
 * Set of all allowed market identifiers (for validation).
 */
export const ALLOWED_MARKETS = new Set(MARKET_COLUMNS.map(m => m.market));

/**
 * List of all market identifiers in order.
 */
export const ALL_MARKETS = MARKET_COLUMNS.map(m => m.market);

File: bet_dashboard/backend/core/market_config.py
"""
Single source of truth for market configuration in the backend.
Reorder entries here to change market order.
"""

from typing import NamedTuple


class MarketDef(NamedTuple):
    """Market definition with all required fields."""

    market: str  # Market identifier (e.g., '1', 'X', 'Over 2.5')
    cons_key: str  # Key for consensus column in DataFrame
    odds_key: str  # Key for odds column in DataFrame
    market_type: str  # Market type for categorization


# Market definitions in order
MARKET_DEFINITIONS = [
    MarketDef("1", "cons_home", "odds_home", "result"),
    MarketDef("X", "cons_draw", "odds_draw", "result"),
    MarketDef("2", "cons_away", "odds_away", "result"),
    MarketDef("Over 2.5", "cons_over_25", "odds_over_25", "over_under_25"),
    MarketDef("Under 2.5", "cons_under_25", "odds_under_25", "over_under_25"),
    MarketDef("BTTS Yes", "cons_btts_yes", "odds_btts_yes", "btts"),
    MarketDef("BTTS No", "cons_btts_no", "odds_btts_no", "btts"),
    MarketDef("Over 0.5", "cons_over_05", "odds_over_05", "over_under_05"),
    MarketDef("Under 0.5", "cons_under_05", "odds_under_05", "over_under_05"),
    MarketDef("Over 1.5", "cons_over_15", "odds_over_15", "over_under_15"),
    MarketDef("Under 1.5", "cons_under_15", "odds_under_15", "over_under_15"),
    MarketDef("Over 3.5", "cons_over_35", "odds_over_35", "over_under_35"),
    MarketDef("Under 3.5", "cons_under_35", "odds_under_35", "over_under_35"),
    MarketDef("Over 4.5", "cons_over_45", "odds_over_45", "over_under_45"),
    MarketDef("Under 4.5", "cons_under_45", "odds_under_45", "over_under_45"),
    MarketDef("1X", "cons_dc_1x", "odds_dc_1x", "double_chance"),
    MarketDef("12", "cons_dc_12", "odds_dc_12", "double_chance"),
    MarketDef("X2", "cons_dc_x2", "odds_dc_x2", "double_chance"),
]

# Set of allowed market identifiers for validation
ALLOWED_MARKETS = {m.market for m in MARKET_DEFINITIONS}

# List of all market identifiers in order
ALL_MARKETS = [m.market for m in MARKET_DEFINITIONS]

# Consensus column names
CONSENSUS_COLUMNS = [m.cons_key for m in MARKET_DEFINITIONS]

# Odds column names
ODDS_COLUMNS = [m.odds_key for m in MARKET_DEFINITIONS]

# Map market identifier to its definition
MARKET_BY_ID = {m.market: m for m in MARKET_DEFINITIONS}
